packages/plugin-legacy/src/index.ts

```diff
@@ -148,11 +143,13 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
 
       // legacy bundle
       if (legacyPolyfills.size || genDynamicFallback) {
-        if (!legacyPolyfills.has('es.promise')) {
-          // check if the target needs Promise polyfill because SystemJS relies
-          // on it
-          await detectPolyfills(`Promise.resolve()`, targets, legacyPolyfills)
-        }
+        // check if the target needs Promise polyfill because SystemJS relies on it
+        // https://github.com/systemjs/systemjs#ie11-support
+        await detectPolyfills(
+          `Promise.resolve(); Promise.all();`,
+          targets,
+          legacyPolyfills
+        )
```

Comment:
Polyfills other than `core-js/modules/es.promise` may be added for `Promise.resolve`/`Promise.all` so the `if` condition is not correct.